llama-index-integrations/vector_stores/llama-index-vector-stores-awsdocdb/llama_index/vector_stores/awsdocdb/awsdocdb.py

```diff
@@ -0,0 +1,305 @@
+
+        Args:
+            docdb_client: A DocumentDB client.
+            db_name: A DocumentDB database name.
+            collection_name: A DocumentDB collection name.
+            id_key: The data field to use as the id.
+            embedding_key: A DocumentDB field that will contain
+            the embedding for each document.
+            text_key: A DocumentDB field that will contain the text for each document.
+            metadata_key: A DocumentDB field that will contain
+            the metadata for each document.
+            insert_kwargs: The kwargs used during `insert`.
+        """
+        import_err_msg = "`pymongo` package not found, please run `pip install pymongo`"
+        try:
+            from pymongo import MongoClient
+        except ImportError:
+            raise ImportError(import_err_msg)
+
+        if docdb_client is not None:
+            self._docdb_client = cast(MongoClient, docdb_client)
+        else:
+            raise ValueError("Must specify connection string to DocumentDB instance ")
+        self._similarity_score = similarity_score
+        self._collection = self._docdb_client[db_name][collection_name]
+        self._embedding_key = embedding_key
+        self._id_key = id_key
+        self._text_key = text_key
+        self._metadata_key = metadata_key
+        self._insert_kwargs = insert_kwargs or {}
+        self._index_crud = DocDbIndex(index_name, self._embedding_key, self._collection)
+
+    def add(
+        self,
+        nodes: List[BaseNode],
+        **add_kwargs: Any,
+    ) -> List[str]:
+        """Add nodes to index.
+
+        Args:
+            nodes: List[BaseNode]: list of nodes with embeddings
+
+        Returns:
+            A List of ids for successfully added nodes.
+
+        """
+        ids = []
+        data_to_insert = []
+        for node in nodes:
+            metadata = node_to_metadata_dict(
+                node, remove_text=True, flat_metadata=self.flat_metadata
+            )
+
+            entry = {
+                self._id_key: node.node_id,
+                self._embedding_key: node.get_embedding(),
+                self._text_key: node.get_content(metadata_mode=MetadataMode.NONE) or "",
+                self._metadata_key: metadata,
+            }
+            data_to_insert.append(entry)
+            ids.append(node.node_id)
+        logger.debug("Inserting data into DocumentDB: %s", data_to_insert)
+        insert_result = self._collection.insert_many(
+            data_to_insert, **self._insert_kwargs
+        )
+        logger.debug("Result of insert: %s", insert_result)
+        return ids
+
+    def delete(self, doc_id: str, **delete_kwargs: Any) -> None:
+        """
+        Delete nodes using by id.
+
+        Args:
+            doc_id (str): The doc_id of the document to delete.
+
+        """
+        if doc_id is None:
+            raise ValueError("No document id provided to delete.")
+        self._collection.delete_one({"id": doc_id})
```

Comment:
This is actually intended to be `delete(ref_doc_id)` -- where any node with `ref_doc_id == ref_doc_id` in its metadata is deleted

Is this possible here?

Reply:
It should be possible. I think we may have misunderstood this method.

Reply:
looks better, thanks!